src/indexed_crate.rs

```diff
@@ -41,6 +41,218 @@ pub struct IndexedCrate<'a> {
     pub(crate) manually_inlined_builtin_traits: HashMap<Id, Item>,
 }
 
+#[cfg(feature = "rayon")]
+use rayon::prelude::*;
+
+/// Map a Key to a List (Vec) of values
+struct MapList<K, V>(HashMap<K, Vec<V>>);
+
+#[cfg(feature = "rayon")]
+impl<K: std::cmp::Eq + std::hash::Hash + Send, V: Send> FromParallelIterator<(K, V)>
+    for MapList<K, V>
+{
+    #[inline]
+    fn from_par_iter<I>(par_iter: I) -> Self
+    where
+        I: IntoParallelIterator<Item = (K, V)>,
+    {
+        par_iter
+            .into_par_iter()
+            .fold(Self::new, |mut map, (key, value)| {
+                map.insert(key, value);
+                map
+            })
+            .reduce(Self::new, |l, mut r| {
+                r.merge(l);
```

Comment:
Interesting that we're merging to the right here. The reasoning behind this is probably worth explaining with a code comment.

Reply:
Turns out merge left is faster, my intuition was wrong on this one XD, left a comment so it isn't swapped without measuring